tests/network/vmi_istio.go

```diff
@@ -306,15 +310,16 @@ var _ = SIGDescribe("[Serial] Istio", func() {
 						vmiPorts = []v1.Port{}
 					})
 					DescribeTable("client outside mesh should NOT reach VMI HTTP server", func(targetPort int) {
-						Expect(checkVMIReachability(vmi, targetPort)).NotTo(Succeed())
+						_, err = checkVMIReachability(vmi, targetPort)
+						Expect(err).To(HaveOccurred())
 					},
 						Entry("on service declared port on VMI with no explicit ports", svcDeclaredTestPort),
 						Entry("on service undeclared port on VMI with no explicit ports", svcUndeclaredTestPort),
 					)
 				})
 			})
 		})
-		Describe("Outbound traffic", func() {
+		Describe("Outbound traffic", Ordered, func() {
```

Comment:
> I think we should not use Ordered
Then IIUC we can't use beforeAll/AfterAll

> one that fails stops all
not necessarily, failure in BeforeAll fail all specs of course, but if one (for example) ping fails in a spec, it doesn't affect subsequent spect, right?

> and it can't be paralleled
Not a biggy, you can have ordered containers running in parallel (each sequentially).
Since the specs run way quicker without repeating setup, IMO worth it

Reply:
> > I think we should not use Ordered
> 
> Then IIUC we can't use beforeAll/AfterAll

True, but you have BeforeSuite for that

> 
> > one that fails stops all
> 
> not necessarily, failure in BeforeAll fail all specs of course, but if one (for example) ping fails in a spec, it doesn't affect subsequent spect, right?

not sure "When a spec in an Ordered container fails, all subsequent specs in the container are skipped."
best to try to fail a `It`
https://onsi.github.io/ginkgo/MIGRATING_TO_V2

> 
> > and it can't be paralleled
> 
> Not a biggy, you can have ordered containers running in parallel (each sequentially). Since the specs run way quicker without repeating setup, IMO worth it

Imho, generally speaking on the whole suite / network suite it isn't right to do so, especially when you have alternative

Reply:
> not sure "When a spec in an Ordered container fails, all subsequent specs in the container are skipped."
best to try to fail a It
https://onsi.github.io/ginkgo/MIGRATING_TO_V2

Yep, you're right. This is admittedly bad

Reply:
> True, but you have BeforeSuite for that

I'm not sure how is BeforeSuite an alternative, Ginkgo allows only one BeforeSuite node, creating all VMs in all configurations required by all the tests in a suite doesn't seems like a way to go.
Unless we split the tests into many suites

Reply:
Gotcha, yep, that may be a way forward